main.cpp

```diff
@@ -423,6 +388,69 @@ IntType& IntType::divide( int lhs )
     return *this;
 }
 
+IntType& IntType::pow(int i)
+{
+    return powInternal(i);
+}
+
+IntType& IntType::pow(const IntType& it)
+{
+    return powInternal(static_cast<int>(it));
+}
+
+IntType& IntType::pow(const FloatType& ft)
+{
+    return powInternal(static_cast<int>(ft));
+}
+
+IntType& IntType::pow(const DoubleType& dt)
+{
+    return powInternal(static_cast<int>(dt));
+}
+
+IntType& IntType::powInternal(const int num)
+{
+    *value = static_cast<int>(std::pow( *value, num ));
+    return *this;
+}
+
+struct Point
+{
+    Point(float ft1, float ft2) : x(ft1), y(ft2) {}
+
+    Point& multiply(float m)
+    {
+        x *= m;
+        y *= m;
+        return *this;
+    }
+    Point& multiply(const IntType& m)
+    {
+        x *= m;
+        y *= m;
+        return *this;
+    }
+    Point& multiply(const FloatType& m)
+    {
+        x *= m;
+        y *= m;
+        return *this;
+    }
+    Point& multiply(const DoubleType& m)
+    {
+        x *= static_cast<float>(m);
+        y *= static_cast<float>(m);
+        return *this;
+    }
```

Comment:
each of these should be doing this internally:
```
return mulitply( static_cast<float>(m) );
```
It's important to identify repetition in code and figure out how to eliminate it.
- [ ] revise